numpy/core/include/numpy/npy_3kcompat.h

```diff
@@ -191,7 +191,7 @@ npy_PyFile_Dup2(PyObject *file, char *mode, npy_off_t *orig_pos)
     if (ret == NULL) {
         return NULL;
     }
-    fd2 = PyNumber_AsSsize_t(ret, NULL);
+    fd2 = (int) PyNumber_AsSsize_t(ret, NULL);
```

Comment:
I guess we should really have a function for this that could also check that the value is in range. Might make a comment so we can find this later if we want to do a proper fix.